### DISCUSSION TITLE
assigning x-model to an expression

### DISCUSSION THREAD

**Discussion by anjanesh** [alpinejs/alpine#3562] [1. Help]
This seems to work :

```
x-model="$store.global_data[grandtotal'] = (1 - ($store.global_data['discount'] / 100)) * $store.global_data['subtotal']"
```

I was wondering how assigning LHS to RHS is resulting in the value of LHS itself and not `true`.

**Comment by ekwoka** [comment]
That's not an expression.

that's an assignment target.

`$store.global_data['grandtotal']` is EXACTLY the same as `$store.global_data.grandtotal` which may be where your confusion is coming from.
